pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -127,24 +126,39 @@ func ctrStop(containerID string, force bool) error {
 	return nil
 }
 
-// ctrRm remove an existing container
-func ctrRm(containerID string) error {
+// ctrRm remove an existing container. If silent is true, then operation failures are ignored and no error is returned
+func ctrRm(containerID string, silent bool) error {
 	log.Infof("ctrRm %s\n", containerID)
 
 	container, err := loadContainer(containerID)
 	if err != nil {
-		return err
+		log.Errorf("ctrRm: exception while loading container: %v. %v", containerID, err.Error())
+		if !silent {
+			return fmt.Errorf("ctrRm: exception while loading container: %v. %v", containerID, err.Error())
+		}
 	}
 	if container == nil {
 		return nil
 	}
 	if err := container.Delete(ctrdCtx); err != nil {
 		log.Errorf("Unable to delete container: %v. %v", containerID, err.Error())
-		return fmt.Errorf("cleanUpContainer: Unable to delete container: %v. %v", containerID, err.Error())
+		if !silent {
+			return fmt.Errorf("cleanUpContainer: Unable to delete container: %v. %v", containerID, err.Error())
+		}
+
+	}
+	snapshotter := ctrdClient.SnapshotService(defaultSnapshotter)
+	if err = snapshotter.Remove(ctrdCtx, getSnapshotName(containerID)); err != nil {
```

Comment:
Why was this necessary? The garbage collector should see a snapshot that is not used by any parent (i.e. container) and thus delete it.

Reply:
Yeah, that was the assumption, but I noticed the snapshot were still present even after calling `container.Delete()` so added a fix where I'm explicitly deleting the snapshot.

Reply:
I have been digging into how the GC works on containerd, so I will see if anything comes up. I am guessing it cannot hurt to delete it, but what happens if it is GC-ed and you try to delete it? Do we get an error?

Also, can you try to delete container without deleting snapshot, and seeing what labels are on the snapshot? Labels affect GC.

Reply:
I checked on the device that after we called `container.Delete()` containerd's GC didn't clean-up the snapshot. 

>what happens if it is GC-ed and you try to delete it? Do we get an error?

After I deleted an app through eve, I waited for an hour to see if GC deletes the snapshot, it didn't